Add large file hash instruction

subdan opened the pull request:
Added large file hash instruction to README file.
Added `init` to Digest protocol to allow initialization from metatype.

Commit:
Add large file hash instruction to README

Commit:
Add init to Digest protocol

modestman commented on README.md:
````diff
@@ -66,3 +66,110 @@ $ wget https://github.com/apple/swift/raw/master/utils/gyb
+
+![Screenshot](./scr.png)
+
+```
````
добавь ```swift

modestman commented on README.md:
```diff
@@ -66,3 +66,110 @@ $ wget https://github.com/apple/swift/raw/master/utils/gyb
+					dispatchIO.close()
+
+					completionQueue.async {
+                        completion(.success(hashFunction.finalize()))
```
табуляция съехала

modestman commented on README.md:
```diff
@@ -66,3 +66,110 @@ $ wget https://github.com/apple/swift/raw/master/utils/gyb
+
+				guard let data = data else { return }
+
+				if data.isEmpty == false {
```
`data.isEmpty` - Optional? 
Почему здесь сравнение с `false`, а не `!data.isEmpty`

subdan replied:
Сама data Optional. Поэтому добавлен guard.
Это одно и то же. Просто == false легче заметить чем узкий символ «!».

modestman reviewed (commented):
Предлагаю разбить функцию `calculateHash` на несколько функций поменьше, или вынести `readNextChunk` в скоуп класса.

Commit:
Update large file hash instruction in README

Move creation of FileHandle and DispatchIO to init.
Add closing of FileHandle when DispatchIO closes a channel.

subdan commented:
> Предлагаю разбить функцию `calculateHash` на несколько функций поменьше, или вынести `readNextChunk` в скоуп класса.

Вынес создание `FileHandle` и `DispatchIO` в `init`. Дополнительно исправил проблему, из-за которой после вычисления хеш-суммы файла он не закрывался, что приводило к тому, что нельзя вычислить хеш-сумму нескольких файлов.

Alexander-Ignition commented on GoldenKey/CommonCrypto/Digest.swift:
```diff
@@ -11,6 +11,8 @@ import CommonCrypto.CommonDigest
 
 public protocol Digest {
 
+    init()
```
`Digest.swift` is generated from `Digest.swift.gyb`

Alexander-Ignition commented on README.md:
````diff
@@ -66,3 +66,118 @@ $ wget https://github.com/apple/swift/raw/master/utils/gyb
+```swift
+final class FileHash {
+    
+    private let queue = DispatchQueue(
````
`queue` cannot be set externally

subdan replied:
What do you mean?

subdan replied:
I've added ability to set work queue in `init`.

Commit:
Add init to Digest.swift.gyb

Commit:
Update large file hash instruction in README

Add ability to set work queue.
Finalize on work queue instead of completion queue.
